browser/ui/views/toolbar/brave_toolbar_view.cc

```diff
@@ -0,0 +1,206 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/toolbar/brave_toolbar_view.h"
+
+#include "brave/browser/ui/views/toolbar/bookmark_button.h"
+#include "chrome/browser/ui/views/toolbar/toolbar_view.h"
+#include "chrome/browser/extensions/extension_util.h"
+#include "chrome/browser/ui/bookmarks/bookmark_bubble_sign_in_delegate.h"
+#include "chrome/browser/ui/layout_constants.h"
+#include "chrome/browser/ui/views/bookmarks/bookmark_bubble_view.h"
+#include "chrome/browser/ui/views/toolbar/browser_app_menu_button.h"
+#include "chrome/browser/ui/views/toolbar/home_button.h"
+#include "chrome/browser/ui/views/toolbar/reload_button.h"
+#include "chrome/browser/ui/views/toolbar/toolbar_button.h"
+#include "ui/base/material_design/material_design_controller.h"
+
+namespace {
+
+int GetToolbarHorizontalPadding() {
+  // In the touch-optimized UI, we don't use any horizontal paddings; the back
+  // button starts from the beginning of the view, and the app menu button ends
+  // at the end of the view.
+  return ui::MaterialDesignController::IsTouchOptimizedUiEnabled() ? 0 : 8;
+}
+
+}  // namespace
+
+void BraveToolbarView::Init() {
+  ToolbarView::Init();
+
+  // Only location bar in non-normal mode
+  if (!is_display_mode_normal()) {
+    return;
+  }
+
+  bookmark_ = new BookmarkButton(this);
```

Comment:
Don't we need to apply upstream's button visibility condition like below?
```
oid LocationBarView::UpdateBookmarkStarVisibility() {
  if (star_view_) {
    star_view_->SetVisible(
        browser_defaults::bookmarks_enabled && !is_popup_mode_ &&
        !GetToolbarModel()->input_in_progress() &&
        edit_bookmarks_enabled_.GetValue() &&
        !IsBookmarkStarHiddenByExtension());
  }
}
```

Reply:
@simonhong I had considered it but I didn't think we needed to because:

- `popup_mode`: this is taken care of in the ToolbarView which will only render the `LocationBarView` when `popup_mode = true`. Since the `BookmarkButton` isn't in the `LocationBarView` anymore, it doesn't need to consider it anymore.
- `input_in_progress`: again, this only makes sense when the bookmark button is inside the location bar, which it isn't anymore
- `IsBookmarkStarHiddenByExtension`: I believe that is for the following extension, which is being discontinued. I don't believe it's an officially supported extension option, but I could be wrong?https://chrome.google.com/webstore/detail/bookmark-manager/gmlllbghnfkpflemihljekbapjopfjik
- `bookmarks_enabled` and `edit_bookmarks_enabled_`: brave-core will always have bookmarks enabled. However, chromium's "Guest Mode" windows do not allow bookmarking. According to the Product team, I believe we won't have Guest Mode windows. It may be worth supporting this just in case, but I was trying to avoid having conditional layout for the button existence.

Let me know if you have any thoughts on those items @simonhong

Reply:
@petemill , got it. Thanks for kind explanation!

Reply:
Now caters to `bookmarks_enabled` and `edit_bookmarks_enabled_` for guest windows.